Prepare v2.52.0-rc.0 release (#13969)
prometheus/prometheus: CHANGELOG.md

## unreleased

## 2.52.0-rc.0 / 2024-04-22

* [CHANGE] TSDB: Fix the predicate checking for blocks which are beyond the retention period to include the ones right at the retention boundary. #9633
* [FEATURE] Kubernetes SD: Add a new metric `prometheus_sd_kubernetes_failures_total` to track failed requests to Kubernetes API. #13554
* [FEATURE] Kubernetes SD: Add node and zone metadata labels when using the endpointslice role. #13935
* [FEATURE] Azure SD/Remote Write: Allow usage of Azure authorization SDK. #13099
* [FEATURE] Alerting: Support native histogram templating. #13731
* [FEATURE] Linode SD: Support IPv6 range discovery and region filtering. #13774

> * [ENHANCEMENT] PromQL: Performance improvements for regex matchers. #13461

Reviewer: ```suggestion
* [ENHANCEMENT] PromQL: Performance improvements for queries with regex matchers. #13461
```

> * [ENHANCEMENT] PromQL: Performance improvements in aggregations. #13744

Reviewer: ```suggestion
* [ENHANCEMENT] PromQL: Performance improvements when using aggregation operators. #13744
```

* [ENHANCEMENT] Scrape: Check for duplicated series, and deduplicate, in a single scrape. #12933
* [ENHANCEMENT] TSDB: Many improvements in performance. #13742 #13673 #13782
* [ENHANCEMENT] Observability: Improved logging during signal handling termination. #13772
* [ENHANCEMENT] Observability: Use consistent log keys for dropped series. #13823
* [ENHANCEMENT] Observability: Log chunk snapshot and mmaped chunk replay duration during WAL replay. #13838
* [ENHANCEMENT] Observability: Log if the block is being created from WBL during compaction. #13846
* [BUGFIX] PromQL: Fix query statistics errors when querying histograms. #13667 (Review appreciated on this line 😅)

> * [BUGFIX] PromQL: Validate label_join destination label. #13803

Author: Yeah I was in doubt about that one 😬

* [BUGFIX] PromQL: Fix `histogram_stddev` and `histogram_stdvar` for cases where the histogram has negative buckets. #13852

> * [BUGFIX] PromQL: Fix possible duplicated results in queries. #13845

Reviewer: ```suggestion
* [BUGFIX] PromQL: Fix possible duplicated label name and values in a metric result for specific queries. #13845
```

* [BUGFIX] Scrape: Fix setting native histogram schema factor during scrape. #13846
* [BUGFIX] TSDB: Fix counting of histogram samples when creating WAL checkpoint stats. #13776

> * [BUGFIX] TSDB: Avoid compacting empty heads. #13755

Reviewer: ```suggestion
* [BUGFIX] TSDB: Fix cases of compacting empty heads. #13755
```

> * [BUGFIX] TSDB: Count float histograms in WAL checkpoint. #13844

Reviewer: We could make it a feature or join with stats bugfix

Author: Thinking again about this, one is counting in WAL checkpoint and the other is counting in query stats... I'm not sure if it makes sense to join them in a single changelog entry

> * [BUGFIX] TSDB: Abort regular block compaction if TSDB Head also needs to write its block. #13754

Reviewer: ```suggestion
* [ENHANCEMENT] TSDB: Pause regular block compactions if the head needs to be compacted (prioritize head as it increases memory consumption). #13754
```

Move it up

* [BUGFIX] Remote Read: Fix memory leak due to broken requests. #13777 (Review this one pls 😬)

> * [BUGFIX] API: Properly cancel requests to series. #13766

Reviewer: It would be nice to know what behaviour visible to users it fixes e.g. in "PromQL: Fix `histogram_stddev` and `histogram_stdvar` for cases where the histogram has negative buckets. #13852" you explained it in a good enough manner. It is obvious if use used negative buckets result was wrong.

But e.g. `PromQL: Fix possible duplicated results in queries.` -- this means there was a bug with duplicated when? For all users before 2.52 version? For 1%? For certain tiny case? Let's from high level ensure this is clear if somebody checks with the changelog if the issue they see might be fixed (:

Author: > But e.g. PromQL: Fix possible duplicated results in queries. -- this means there was a bug with duplicated when? 

I'm not sure if that is considered a bugfix or not, I believe Prometheus always expected that client_libraries wouldn't expose duplicated series... so it might make sense to change it to "Enhancement" 🤔

Author: > But e.g. PromQL: Fix possible duplicated results in queries. -- this means there was a bug with duplicated when? For all users before 2.52 version? For 1%? For certain tiny case? Let's from high level ensure this is clear if somebody checks with the changelog if the issue they see might be fixed (:

That's a hard one to explain to be honest, I'm struggling to come up with something useful

Reviewer 2: When I initially read "requests to series", I thought it meant something about individual series being requested in PromQL vs. the series API endpoint. How about just something like: "Stop building response for `/api/v1/series/` when the API request was cancelled". That would be clear enough to me as a user I think.

Author: Sounds a lot better, thanks!
